drivers/sensor/icp101xx/icp101xx.c

```diff
@@ -0,0 +1,235 @@
+			return 7;
+		break;
+		case ICP101XX_MEAS_LOW_NOISE_T_FIRST:
+		case ICP101XX_MEAS_LOW_NOISE_P_FIRST:
+			return 24;
+		break;
+		default:
+		case ICP101XX_MEAS_ULTRA_LOW_NOISE_T_FIRST:
+		case ICP101XX_MEAS_ULTRA_LOW_NOISE_P_FIRST:
+			return 95;
+		break;
+	}
+}
+
+static uint8_t get_conversion_ms(enum icp101xx_meas mode)
+{
+	switch (mode)
+	{
+		case ICP101XX_MEAS_LOW_POWER_T_FIRST:
+		case ICP101XX_MEAS_LOW_POWER_P_FIRST:
+			return 1;
+		break;
+		case ICP101XX_MEAS_NORMAL_T_FIRST:
+		case ICP101XX_MEAS_NORMAL_P_FIRST:
+			return 5;
+		break;
+		case ICP101XX_MEAS_LOW_NOISE_T_FIRST:
+		case ICP101XX_MEAS_LOW_NOISE_P_FIRST:
+			return 20;
+		break;
+		default:
+		case ICP101XX_MEAS_ULTRA_LOW_NOISE_T_FIRST:
+		case ICP101XX_MEAS_ULTRA_LOW_NOISE_P_FIRST:
+			return 80;
+		break;
+	}
+}
+
+#define ATMOSPHERICAL_PRESSURE_KPA 101.325
+#define TO_KELVIN(temp_C) (273.15 + temp_C)
+#define HEIGHT_TO_PRESSURE_COEFF 0.03424 // M*g/R = (0,0289644 * 9,80665 / 8,31432)
+
+#define PRESSURE_TO_HEIGHT_COEFF 29.27127 // R / (M*g) = 8,31432 / (0,0289644 * 9,80665)
+#define LOG_ATMOSPHERICAL_PRESSURE 4.61833 // ln(101.325)
+
+float convertToHeight(float pressure_kp, float temperature_C)
+{
+	return PRESSURE_TO_HEIGHT_COEFF * TO_KELVIN(temperature_C) * (LOG_ATMOSPHERICAL_PRESSURE - log(pressure_kp));
+}
+
+static int icp101xx_attr_set(const struct device *dev,
+			     enum sensor_channel chan,
+			     enum sensor_attribute attr,
+			     const struct sensor_value *val)
+{
+	int err = 0;
+	icp101xx_data* data = (icp101xx_data*)dev->data;
+	__ASSERT_NO_MSG(val != NULL);
+	if(chan == SENSOR_CHAN_PRESS)
+	{
+		if (attr == SENSOR_ATTR_CONFIGURATION) {
+			if ((val->val1 >= ICP101XX_MEAS_LOW_POWER_T_FIRST)&&(val->val1 <= ICP101XX_MEAS_ULTRA_LOW_NOISE_P_FIRST)) {
+				data->icp_device.measurement_mode = val->val1;
+			} else {
+				LOG_ERR("Not supported ATTR value");
+				return -EINVAL;
+			}
+			
+		} else {
+			LOG_ERR("Not supported ATTR");
+			return -EINVAL;
+		}
+	};
+	return err;
+}
+
+static int icp101xx_sample_fetch(const struct device *dev, const enum sensor_channel chan)
+{
+	icp101xx_data *data = (icp101xx_data*)dev->data;
+	int rc = 0;
+	uint64_t timeout;
+
+	if (!(chan == SENSOR_CHAN_AMBIENT_TEMP || chan == SENSOR_CHAN_PRESS || SENSOR_CHAN_ALTITUDE ||
+		chan == SENSOR_CHAN_ALL)) {
+		return -ENOTSUP;
+	}
+	rc = inv_icp101xx_enable_sensor(&data->icp_device, 1);
+	// Compute timeout for the measure
+	timeout = k_uptime_get() + get_timeout_ms(data->icp_device.measurement_mode);
+	// Initial sleep waiting the sensor proceeds with the measure
+	k_sleep(K_MSEC(get_conversion_ms(data->icp_device.measurement_mode)));
+	do {
+		k_sleep(K_USEC(200));
+		rc = inv_icp101xx_get_data(&data->icp_device, &(data->raw_pressure), &(data->raw_temperature), &(data->pressure), &(data->temperature));
+	} while ((rc != 0) && (k_uptime_get() <= timeout));
+	// Zephyr expects kPa while ICP101xx returns Pa
+	data->pressure = data->pressure/1000;
```

Comment:
Maybe this calcul can be done in channel_get ? For motion driver, this is the logic I followed, floating computing are done after calling channel_get API()

Reply:
If you like so :)